services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/hakemus/HakemusServiceITest.kt

```diff
@@ -72,380 +72,749 @@ class HakemusServiceITest(
 ) : IntegrationTest() {
 
     @Nested
-    inner class HakemusResponse {
-        @Test
-        fun `when application does not exist should throw`() {
-            assertThat(applicationRepository.findAll()).isEmpty()
+    inner class Common {
```

Comment:
When doing refactoring that changes entire files, it would be nice to do the refactoring separately without any other changes first. This reduces the changes per PR and makes it easy to see what meaningful changes there are. Now the only way to see it is to compare the files line-by-line.

Reply:
Sorry, I am still struggling sometimes to create a decent size PRs. I will try to do better in the future.